Examples: App with RBAC (#242)
grafana/grafana-plugin-examples: examples/app-with-rbac/README.md

      ]
    },
    "grants": ["Viewer"]
  }
]
```

### Protect includes

To protect you frontend pages behind an action check, add `action` to the include definitions of your `plugin.json` file:

```json
"includes": [
  {
    "type": "page",
    "name": "Research documents",
    "path": "/a/%PLUGIN_ID%/research-docs",
    "action": "grafana-appwithrbac-app.papers:read",
    "addToNav": true,
    "defaultNav": false
  },
  {
    "type": "page",
    "name": "Patents",
    "path": "/a/%PLUGIN_ID%/patents",
    "action": "grafana-appwithrbac-app.patents:read",
    "addToNav": true,
    "defaultNav": false
  }
]
```

### Protect routes

This is not demonstrated in this plugin but if you want to protect your proxied routes behind an action check, add `reqAction` to the route definitions of your `plugin.json` file:

```json
"routes": [
  {
    "path": "api/external/patents",
    "method": "*",
    "reqAction": "grafana-appwithrbac-app.patents:read",
    "url": "{{ .JsonData.backendUrl }}/api/external/patents",
    "headers": [
      {
        "name": "Authorization",
        "content": "{{ .SecureJsonData.backendApiToken }}"
      }
    ]
  }
]
```

> ### Protect you plugin backend resources

Reviewer: ```suggestion
### Protect plugin backend resources
```

    "path": "/a/%PLUGIN_ID%/research-docs",
    "action": "grafana-appwithrbac-app.papers:read",
    "addToNav": true,
    "defaultNav": false
  },
  {
    "type": "page",
    "name": "Patents",
    "path": "/a/%PLUGIN_ID%/patents",
    "action": "grafana-appwithrbac-app.patents:read",
    "addToNav": true,
    "defaultNav": false
  }
]
```

### Protect routes

If you want to protect your proxied routes behind an action check, add `reqAction` to the route definitions of your `plugin.json` file. For example:

```json
"routes": [
  {
    "path": "api/external/patents",
    "method": "*",
    "reqAction": "grafana-appwithrbac-app.patents:read",
    "url": "{{ .JsonData.backendUrl }}/api/external/patents",
    "headers": [
      {
        "name": "Authorization",
        "content": "{{ .SecureJsonData.backendApiToken }}"
      }
    ]
  }
]
```

Note that this feature is not demonstrated in this plugin.

### Protect plugin backend resources

If your backend is exposing resources, you can also protect them behind an action check.

To do so, activate two additional features:

- `externalServiceAccounts` - to get a service account to query Grafana's user permissions.

> - `idForwarding` - to receive id tokens identifying the requester (user/service account).

Author: @josmperez does this phrasing work for you?

> - `externalServiceAccounts` - to get a managed service account to query Grafana's users permissions.
> - `idForwarding` - to receive id tokens identifying the requester (user/service account).

Author: ```suggestion
- `externalServiceAccounts` - To obtain a managed service account to retrieve Grafana users' permissions.
- `idForwarding` - To obtain ID tokens that identify the requester (user or service account).
```

> - `idForwarding` - To obtain ID tokens that identify the requester (user or service account).

Author: ```suggestion
- `idForwarding` - To obtain an ID token that identify the requester (user or service account).
```
